src/model/pwrmgmt.go

```diff
@@ -23,15 +25,76 @@ var scalProfilesMap = map[string]ScalProfiles{
 type CpuGovernanceRule struct {
 	CPUs    string `yaml:"cpus"`
 	ScalGov string `yaml:"scaling_governor"`
+	MinFreq string `yaml:"min_freq"`
+	MaxFreq string `yaml:"max_freq"`
 }
 
 func (c CpuGovernanceRule) Validate() error {
-	if _, ok := scalProfilesMap[c.ScalGov]; !ok {
+	if _, err := cpulists.Parse(c.CPUs); err != nil {
+		return err
+	}
+	if _, ok := scalProfilesMap[c.ScalGov]; !ok && c.ScalGov != "" {
 		return fmt.Errorf("invalid cpu scaling governor: %v", c.ScalGov)
 	}
-	_, err := cpulists.Parse(c.CPUs)
-	if err != nil {
-		return fmt.Errorf("invalid cpus: %v", err)
+	min, errMin := ParseFreq(c.MinFreq)
+	if errMin != nil {
+		return fmt.Errorf("invalid min_freq: %v", errMin)
+	}
+	max, errMax := ParseFreq(c.MaxFreq)
+	if errMax != nil {
+		return fmt.Errorf("invalid max_freq: %v", errMax)
+	}
+
+	minAndmaxAreSet := (min != -1 && max != -1) && (min != 0 && max != 0)
+	if (max < min) && minAndmaxAreSet {
+		return fmt.Errorf(
+			"max frequency (%d) cannot be less than min frequency (%d)",
+			max, min)
+	}
+	if (min == max) && minAndmaxAreSet {
+		return fmt.Errorf("min and max frequency cannot be the same: %d", min)
 	}
+
 	return nil
 }
+
+func ParseFreq(freq string) (int, error) {
+	if freq == "" {
+		return -1, nil // No frequency limits set, nothing to parse
+	}
+
+	s := strings.ToLower(strings.TrimSpace(freq))
+	if !strings.HasSuffix(s, "hz") {
+		return -1, fmt.Errorf(
+			"invalid format: frequency must end with 'Hz': %s", s)
+	}
+
+	s = strings.TrimSuffix(s, "hz")
+
+	multiplier := 1.0
+	switch {
+	case strings.HasSuffix(s, "g"):
+		multiplier = 1_000_000.0
+		s = strings.TrimSuffix(s, "g")
+	case strings.HasSuffix(s, "m"):
+		multiplier = 1_000.0
+		s = strings.TrimSuffix(s, "m")
+	case strings.HasSuffix(s, "k"):
+		multiplier = 1.0
+		s = strings.TrimSuffix(s, "k")
+	default:
+		multiplier = 0.001 // Default to raw Hz if no suffix is provided
```

Comment:
```suggestion
		multiplier = 0.001 // Default to raw Hz if no suffix remains
```
or `Default to raw Hz if no metric prefix is provided`

Reply:
Liked the second suggestion: 168d771